### PR TITLE
Update vector.rst in the Cookbook to include selectbyexpression (#1028)

### PR DESCRIPTION
**Pull request by havatv**
Attempt to address #1028 by including the example suggested by @nyalldawson in #1028.

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 1]
```diff
@@ -105,6 +105,14 @@ To select all the features:
  # Get the active layer (must be a vector layer)
  layer = iface.activeLayer()
  layer.selectAll()
+ 
+To select using an expression:
+
+.. code-block:: python
+
+ # Assumes that the active layer is points.shp file from the QGIS test suite
```
Comment (DelazJ) [inline source/docs/pyqgis_developer_cookbook/vector.rst]: @havatv sorry I'm late but while I'm a bit close to the project I don't know where I could find this dataset so people who just want a code... Reading this chapter, it looks like the data used (I don't know where it's nor whether it exists) contains fields like `name`, `id`, `location_name`... Maybe:
* reuse these fields
* indicate that `Class` and `Heading` are fields (they are, right?)
* or use `field1` and `field2` instead?

Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: You are probably right - I just used @nyalldawson's example code.

Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: I have tried to improve the text by explaining.

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 2]
```diff
@@ -105,6 +105,14 @@ To select all the features:
  # Get the active layer (must be a vector layer)
  layer = iface.activeLayer()
  layer.selectAll()
+ 
+To select using an expression:
+
+.. code-block:: python
+
+ # Get the points.shp file from the QGIS test suite
+ layer = QgsVectorLayer('points.shp'), 'Points', 'ogr')
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: Done.

**Pull request merged**